src/components/ProductsList.jsx

```diff
@@ -1,25 +1,47 @@
 import React, { Component } from "react";
 import { fetchProductData } from "../modules/products";
+import { createOrder } from "../modules/orders";
 
 class ProductsList extends Component {
   state = {
     products: [],
+    orderDetails: {},
   };
 
   componentDidMount = async () => {
     const products = await fetchProductData();
     this.setState({ products: products });
   };
 
+  addToOrder = async (event) => {
+    let productId = event.target.parentElement.dataset.id;
+    let result = await createOrder(productId);
+    this.setState({
+      orderDetails: {
+        id: productId,
+        message: result.message,
+      },
+    });
+  };
+
   render() {
     let productsList;
+
     if (this.state.products !== []) {
       productsList = this.state.products.map((product) => {
         return (
-          <div data-cy={"product-" + product.id} key={product.id}>
+          <div data-cy={"product-" + product.id} data-id={product.id}>
```

Comment:
Why have you changed this? You always need a key when you map over an array in React